In [ ]:
import os
import numpy as np
import nibabel as nib
import matplotlib.pyplot as plt

In [ ]:
no = "2"

In [ ]:
for nos in range(3, 7):
    print("____________________" + str(nos))
    global no, img, img2, xn, yn, zn, d, d2, organ1, organData1, organ2, organData2, organ3, organData3
    no = str(nos)
    img = nib.load('D:/projects/finalProject/nifti/' + no + '/image.nii.gz')
    img2 = nib.load('D:/projects/finalProject/nifti/' + no + '/mask.nii.gz')
    xn = img.shape[0]
    yn = img.shape[1]
    zn = img.shape[2]
    print(img.shape)
    d = img.get_data()
    d2 = img2.get_data()
    writeFile1("D:/projects/finalProject/nifti/" + no + "/image.vtk", "image")
    print("file1 done")
    writeFile("D:/projects/finalProject/nifti/" + no + "/mask.vtk", "mask")
    print("file2 done")
    writeFile2("D:/projects/finalProject/nifti/" + no + "/imageFullRange.vtk", "imageFullRange")
    print("file3 done")
    organ1 = getOrgan(1)
    organData1 = unique(organ1[0])
    print("organ1 done")
    organ2 = getOrgan(2)
    organData2 = unique(organ2[0])
    print("organ2 done")
    organ3 = getOrgan(3)
    organData3 = unique(organ3[0])
    print("organ3 done")

    print("total organ area ({} to {})".format(min(organData1[1], organData2[1], organData3[1]), max(organData1[2], organData2[2], organData3[2])))
    print("(x: {}, {}); (y: {}, {}); (z: {}, {})".format(min(organ1[1], organ2[1], organ3[1]), max(organ1[2], organ2[2], organ3[2]), min(organ1[3], organ2[3], organ3[3]), max(organ1[4], organ2[4], organ3[4]), min(organ1[5], organ2[5], organ3[5]), max(organ1[6], organ2[6], organ3[6])))
    print("organ 1 ({} to {})".format(organData1[1], organData1[2]))
    print("(x: {}, {}); (y: {}, {}); (z: {}, {})".format(organ1[1], organ1[2], organ1[3], organ1[4], organ1[5], organ1[6]))
    print("organ 2 ({} to {})".format(organData2[1], organData2[2]))
    print("(x: {}, {}); (y: {}, {}); (z: {}, {})".format(organ2[1], organ2[2], organ2[3], organ2[4], organ2[5], organ2[6]))
    print("organ 3 ({} to {})".format(organData3[1], organData3[2]))
    print("(x: {}, {}); (y: {}, {}); (z: {}, {})".format(organ3[1], organ3[2], organ3[3], organ3[4], organ3[5], organ3[6]))
    writeOrganArea()
    writeOrganVals(organData1, 1)
    writeOrganVals(organData2, 2)
    writeOrganVals(organData3, 3)

In [ ]:
for nos in range(110, 133):
    print("____________________" + str(nos))
    global no, img, img2, xn, yn, zn, d, d2, organ1, organData1, organ2, organData2, organ3, organData3
    no = str(nos)
    img = nib.load('D:/projects/finalProject/nifti/' + no + '/image.nii.gz')
    img2 = nib.load('D:/projects/finalProject/nifti/' + no + '/mask.nii.gz')
    xn = img.shape[0]
    yn = img.shape[1]
    zn = img.shape[2]
    print(img.shape)
    d = img.get_data()
    d2 = img2.get_data()
    # writeFile1("D:/projects/finalProject/nifti/" + no + "/image.vtk", "image")
    # print("file1 done")
    # writeFile("D:/projects/finalProject/nifti/" + no + "/mask.vtk", "mask")
    # print("file2 done")
    # writeFile2("D:/projects/finalProject/nifti/" + no + "/imageFullRange.vtk", "imageFullRange")
    # print("file3 done")
    organ1 = getOrgan(1)
    organData1 = unique(organ1[0])
    print("organ1 done")
    organ2 = getOrgan(2)
    organData2 = unique(organ2[0])
    print("organ2 done")
    organ3 = getOrgan(3)
    organData3 = unique(organ3[0])
    print("organ3 done")

    print("total organ area ({} to {})".format(min(organData1[1], organData2[1], organData3[1]), max(organData1[2], organData2[2], organData3[2])))
    print("(x: {}, {}); (y: {}, {}); (z: {}, {})".format(min(organ1[1], organ2[1], organ3[1]), max(organ1[2], organ2[2], organ3[2]), min(organ1[3], organ2[3], organ3[3]), max(organ1[4], organ2[4], organ3[4]), min(organ1[5], organ2[5], organ3[5]), max(organ1[6], organ2[6], organ3[6])))
    print("organ 1 ({} to {})".format(organData1[1], organData1[2]))
    print("(x: {}, {}); (y: {}, {}); (z: {}, {})".format(organ1[1], organ1[2], organ1[3], organ1[4], organ1[5], organ1[6]))
    print("organ 2 ({} to {})".format(organData2[1], organData2[2]))
    print("(x: {}, {}); (y: {}, {}); (z: {}, {})".format(organ2[1], organ2[2], organ2[3], organ2[4], organ2[5], organ2[6]))
    print("organ 3 ({} to {})".format(organData3[1], organData3[2]))
    print("(x: {}, {}); (y: {}, {}); (z: {}, {})".format(organ3[1], organ3[2], organ3[3], organ3[4], organ3[5], organ3[6]))
    writeOrganArea()
    writeOrganVals(organData1, 1)
    writeOrganVals(organData2, 2)
    writeOrganVals(organData3, 3)

____________________110
(512, 512, 161)


C:\Users\nrmlg\AppData\Local\Temp\ipykernel_25268\1943520258.py:11: DeprecationWarning: get_data() is deprecated in favor of get_fdata(), which has a more predictable return type. To obtain get_data() behavior going forward, use numpy.asanyarray(img.dataobj).

* deprecated from version: 3.0
* Will raise <class 'nibabel.deprecator.ExpiredDeprecationError'> as of version: 5.0
  d = img.get_data()
C:\Users\nrmlg\AppData\Local\Temp\ipykernel_25268\1943520258.py:12: DeprecationWarning: get_data() is deprecated in favor of get_fdata(), which has a more predictable return type. To obtain get_data() behavior going forward, use numpy.asanyarray(img.dataobj).

* deprecated from version: 3.0
* Will raise <class 'nibabel.deprecator.ExpiredDeprecationError'> as of version: 5.0
  d2 = img2.get_data()


organ1 done
organ2 done
organ3 done
total organ area (-990.0 to 1428.0)
(x: 197, 308); (y: 206, 340); (z: 65, 111)
organ 1 (-892.0 to 1004.0)
(x: 197, 308); (y: 206, 275); (z: 74, 92)
organ 2 (-608.0 to 1279.0)
(x: 227, 283); (y: 289, 337); (z: 65, 84)
organ 3 (-990.0 to 1428.0)
(x: 200, 294); (y: 233, 340); (z: 85, 111)
____________________111
(512, 512, 148)
organ1 done
organ2 done
organ3 done
total organ area (-951.0 to 1430.0)
(x: 191, 305); (y: 224, 353); (z: 61, 96)
organ 1 (-951.0 to 950.0)
(x: 191, 305); (y: 224, 304); (z: 61, 83)
organ 2 (-941.0 to 1397.0)
(x: 217, 280); (y: 307, 353); (z: 64, 87)
organ 3 (-230.0 to 1430.0)
(x: 214, 271); (y: 272, 344); (z: 85, 96)
____________________112
(512, 512, 140)
organ1 done
organ2 done
organ3 done
total organ area (-919.0 to 1488.0)
(x: 204, 326); (y: 210, 384); (z: 50, 78)
organ 1 (-919.0 to 1488.0)
(x: 222, 326); (y: 210, 310); (z: 50, 71)
organ 2 (-708.0 to 1002.0)
(x: 226, 288); (y: 327, 384); (z: 51, 77)
organ 3 (-121.0 to 983.0)

In [ ]:
# example_filename = os.path.join(data_path, 'example4d.nii.gz')
img = nib.load('D:/projects/finalProject/nifti/' + no + '/image.nii.gz')
img2 = nib.load('D:/projects/finalProject/nifti/' + no + '/mask.nii.gz')
xn = img.shape[0]
yn = img.shape[1]
zn = img.shape[2]
img.shape

(512, 512, 125)

In [ ]:
d = img.get_data()
d2 = img2.get_data()

In [ ]:
# write full image
def writeFile1(fileName, scalarDataName):
    data = '''# vtk DataFile Version 2.0\nVolume example\nASCII\nDATASET STRUCTURED_POINTS\nDIMENSIONS {} {} {}\nASPECT_RATIO 0.93 0.93 5\nORIGIN 0 0 0\nPOINT_DATA {}\nSCALARS {} float 1\nLOOKUP_TABLE default\n'''.format(xn, yn, zn, xn * yn * zn, scalarDataName)
    tD = [0 for i in range(xn * yn * zn)]
    tN = 0
    tVN = 0
    for z in range(zn):
        for y in range(yn):
            for x in range(xn):
                tD[tN] = "{:.2f}".format(d[x][y][z])
                tN = tN + 1
    data = data + " ".join(tD)
    f = open(fileName, "w")
    f.write(data)
    f.close()
# writeFile1("D:/projects/finalProject/nifti/" + no + "/image.vtk", "image")

In [ ]:
# write mask
def writeFile(fileName, scalarDataName):
    data = '''# vtk DataFile Version 2.0\nVolume example\nASCII\nDATASET STRUCTURED_POINTS\nDIMENSIONS {} {} {}\nASPECT_RATIO 0.93 0.93 5\nORIGIN 0 0 0\nPOINT_DATA {}\nSCALARS {} float 1\nLOOKUP_TABLE default\n'''.format(xn, yn, zn, xn * yn * zn, scalarDataName)
    tD = [0 for i in range(xn * yn * zn)]
    tN = 0
    tVN = 0
    for z in range(zn):
        for y in range(yn):
            for x in range(xn):
                tD[tN] = "{:.2f}".format(d2[x][y][z])
                tN = tN + 1
    data = data + " ".join(tD)
    f = open(fileName, "w")
    f.write(data)
    f.close()
# writeFile("D:/projects/finalProject/nifti/" + no + "/mask.vtk", "mask")

In [ ]:
# write full in range
def writeFile2(fileName, scalarDataName):
    data = '''# vtk DataFile Version 2.0\nVolume example\nASCII\nDATASET STRUCTURED_POINTS\nDIMENSIONS {} {} {}\nASPECT_RATIO 0.93 0.93 5\nORIGIN 0 0 0\nPOINT_DATA {}\nSCALARS {} float 1\nLOOKUP_TABLE default\n'''.format(xn, yn, zn, xn * yn * zn, scalarDataName)
    tD = [0 for i in range(xn * yn * zn)]
    tN = 0
    tVN = 0
    for z in range(zn):
        for y in range(yn):
            for x in range(xn):
                if d[x][y][z] >= -879 and d[x][y][z] <= 73:
                    tD[tN] = "{:.2f}".format(d[x][y][z])
                else:
                    tD[tN] = "{:.2f}".format(-880)
                tN = tN + 1
    data = data + " ".join(tD)
    f = open(fileName, "w")
    f.write(data)
    f.close()
# writeFile2("D:/projects/finalProject/nifti/" + no + "/imageFullRange.vtk", "imageFullRange")

In [ ]:
def unique(list1):
    uniqueList = [ ]
    tMin = float("inf")
    tMax = float("-inf")
    seqMin = list1[0]
    seqMax = list1[0]
    seqList = [ ]
    for x in list1:
        if x not in uniqueList:
            uniqueList.append(x)
        if x < tMin:
            tMin = x
        if x > tMax:
            tMax = x
        if (seqMin - 1) <= x and (seqMax + 1) >= x and seqMin > x:
            seqMin = x
        elif (seqMin - 1) <= x and (seqMax + 1) >= x and seqMax < x:
            seqMax = x
        elif (seqMin - 1) <= x and (seqMax + 1) >= x:
            continue
        else:
            seqList.append((seqMin, seqMax, seqMax - seqMin))
            seqMin = x
            seqMax = x
    return [uniqueList, tMin, tMax, seqList]
def getOrgan(findFor):
    txMin = xn - 1
    txMax = 0
    tyMin = yn - 1
    tyMax = 0
    tzMin = zn - 1
    tzMax = 0
    tData = [ ]
    for z in range(zn):
        for y in range(yn):
            for x in range(xn):
                if d2[x][y][z] == findFor:
                    tData.append(d[x][y][z])
                if d2[x][y][z] == findFor and txMin > x:
                    txMin = x
                if d2[x][y][z] == findFor and txMax < x:
                    txMax = x
                if d2[x][y][z] == findFor and tyMin > y:
                    tyMin = y
                if d2[x][y][z] == findFor and tyMax < y:
                    tyMax = y
                if d2[x][y][z] == findFor and tzMin > z:
                    tzMin = z
                if d2[x][y][z] == findFor and tzMax < z:
                    tzMax = z
    tData.sort()
    return [tData, txMin, txMax, tyMin, tyMax, tzMin, tzMax]
# organ1 = getOrgan(1)
# organData1 = unique(organ1[0])
# organ2 = getOrgan(2)
# organData2 = unique(organ2[0])
# organ3 = getOrgan(3)
# organData3 = unique(organ3[0])

# print("total organ area ({} to {})".format(min(organData1[1], organData2[1], organData3[1]), max(organData1[2], organData2[2], organData3[2])))
# print("(x: {}, {}); (y: {}, {}); (z: {}, {})".format(min(organ1[1], organ2[1], organ3[1]), max(organ1[2], organ2[2], organ3[2]), min(organ1[3], organ2[3], organ3[3]), max(organ1[4], organ2[4], organ3[4]), min(organ1[5], organ2[5], organ3[5]), max(organ1[6], organ2[6], organ3[6])))
# print("organ 1 ({} to {})".format(organData1[1], organData1[2]))
# print("(x: {}, {}); (y: {}, {}); (z: {}, {})".format(organ1[1], organ1[2], organ1[3], organ1[4], organ1[5], organ1[6]))
# print("organ 2 ({} to {})".format(organData2[1], organData2[2]))
# print("(x: {}, {}); (y: {}, {}); (z: {}, {})".format(organ2[1], organ2[2], organ2[3], organ2[4], organ2[5], organ2[6]))
# print("organ 3 ({} to {})".format(organData3[1], organData3[2]))
# print("(x: {}, {}); (y: {}, {}); (z: {}, {})".format(organ3[1], organ3[2], organ3[3], organ3[4], organ3[5], organ3[6]))

In [ ]:
def writeOrganArea():
    f = open("D:/projects/finalProject/nifti/" + no + "/ranges.txt", "a")
    data = "total organ area ({} to {})\n".format(min(organData1[1], organData2[1], organData3[1]), max(organData1[2], organData2[2], organData3[2]))
    data = data + "(x: {}, {}); (y: {}, {}); (z: {}, {})\n".format(min(organ1[1], organ2[1], organ3[1]), max(organ1[2], organ2[2], organ3[2]), min(organ1[3], organ2[3], organ3[3]), max(organ1[4], organ2[4], organ3[4]), min(organ1[5], organ2[5], organ3[5]), max(organ1[6], organ2[6], organ3[6]))
    data = data + "organ 1 ({} to {})\n".format(organData1[1], organData1[2])
    data = data + "(x: {}, {}); (y: {}, {}); (z: {}, {})\n".format(organ1[1], organ1[2], organ1[3], organ1[4], organ1[5], organ1[6])
    data = data + "organ 2 ({} to {})\n".format(organData2[1], organData2[2])
    data = data + "(x: {}, {}); (y: {}, {}); (z: {}, {})\n".format(organ2[1], organ2[2], organ2[3], organ2[4], organ2[5], organ2[6])
    data = data + "organ 3 ({} to {})\n".format(organData3[1], organData3[2])
    data = data + "(x: {}, {}); (y: {}, {}); (z: {}, {})".format(organ3[1], organ3[2], organ3[3], organ3[4], organ3[5], organ3[6])
    
    f.write(data)
    f.close()
# writeOrganArea()

In [ ]:
def writeOrganVals(d, name):
    f = open("D:/projects/finalProject/nifti/{}/type{}.txt".format(no, name), "a")
    f.write(str(d[3]))
    f.close()
# writeOrganVals(organData3, 3)
# organData1[3]

100

total organ area

(x: 200, 305); (y: 258, 417); (z: 74, 105)

organ 1 (-945.0 to 1359.0)

(x: 200, 305); (y: 258, 335); (z: 75, 94)

organ 2 (-380.0 to 1491.0)

(x: 240, 283); (y: 363, 417); (z: 74, 94)

organ 3 (-937.0 to 4395.0)

(x: 236, 289); (y: 266, 411); (z: 95, 105)

1

total organ area

x 212 305;
y 216 341;
z 40 58

organ 1 (-948.0 to 1016.0)

x 212 305;
y 216 276;
z 44 54

organ 2 (-961.0 to 1209.0)

x 231 287;
y 297 341;
z 40 54

organ 3 (-972.0 to 1176.0)

x 218 267;
y 265 315;
z 54 58

In [ ]:
# total organ area
# 212 305
# 216 341
# 40 58
# (x: 200, 305); (y: 258, 417); (z: 74, 105)

def writeFile3(fileName, scalarDataName):
    data = '''# vtk DataFile Version 2.0\nVolume example\nASCII\nDATASET STRUCTURED_POINTS\nDIMENSIONS {} {} {}\nASPECT_RATIO 0.93 0.93 5\nORIGIN 0 0 0\nPOINT_DATA {}\nSCALARS {} float 1\nLOOKUP_TABLE default\n'''.format(xn, yn, zn, xn * yn * zn, scalarDataName)
    tD = [0 for i in range(xn * yn * zn)]
    tN = 0
    tVN = 0
    for z in range(zn):
        for y in range(yn):
            for x in range(xn):
                if x >= 200 and x <= 305 and y >= 258 and y <= 417 and z >= 74 and z <= 105:
                    tD[tN] = "{:.2f}".format(d[x][y][z])
                else:
                    tD[tN] = "{:.2f}".format(-1161)
                tN = tN + 1
            # print(z, y)
    data = data + " ".join(tD)
    f = open(fileName, "w")
    f.write(data)
    f.close()
# writeFile3("D:/projects/finalProject/nifti/" + no + "/imageBounded.vtk", "imageBounded")

In [ ]:
# print organs
def writeFile4(organ, organMin, fileName, scalarDataName):
    data = '''# vtk DataFile Version 2.0\nVolume example\nASCII\nDATASET STRUCTURED_POINTS\nDIMENSIONS {} {} {}\nASPECT_RATIO 0.93 0.93 5\nORIGIN 0 0 0\nPOINT_DATA {}\nSCALARS {} float 1\nLOOKUP_TABLE default\n'''.format(xn, yn, zn, xn * yn * zn, scalarDataName)
    tD = [0 for i in range(xn * yn * zn)]
    tN = 0
    tVN = 0
    for z in range(zn):
        for y in range(yn):
            for x in range(xn):
                if d2[x][y][z] == organ:
                    tD[tN] = "{:.2f}".format(d[x][y][z])
                else:
                    tD[tN] = "{:.2f}".format(organMin)
                tN = tN + 1
            # print(z, y)
    data = data + " ".join(tD)
    f = open(fileName, "w")
    f.write(data)
    f.close()
# writeFile4(1, organData1[1] - 1, "D:/projects/finalProject/nifti/" + no + "/organ1.vtk", "organ1")
# print("done1")
# writeFile4(2, organData2[1] - 1, "D:/projects/finalProject/nifti/" + no + "/organ2.vtk", "organ2")
# print("done2")
# writeFile4(3, organData3[1] - 1, "D:/projects/finalProject/nifti/" + no + "/organ3.vtk", "organ3")
# print("done3")

In [ ]:
def writeFile4(fileName, scalarDataName):
    data = '''# vtk DataFile Version 2.0\nVolume example\nASCII\nDATASET STRUCTURED_POINTS\nDIMENSIONS {} {} {}\nASPECT_RATIO 0.93 0.93 5\nORIGIN 0 0 0\nPOINT_DATA {}\nSCALARS {} float 1\nLOOKUP_TABLE default\n'''.format(xn, yn, zn, xn * yn * zn, scalarDataName)
    tD = [0 for i in range(xn * yn * zn)]
    tN = 0
    tVN = 0
    for z in range(zn):
        for y in range(yn):
            for x in range(xn):
                if x >= 212 and x <= 305 and y >= 216 and y <= 341 and z >= 40 and z <= 58:
                    tD[tN] = "{:.2f}".format(d[x][y][z])
                else:
                    tD[tN] = "{:.2f}".format(-1161)
                tN = tN + 1
            # print(z, y)
    data = data + " ".join(tD)
    f = open(fileName, "w")
    f.write(data)
    f.close()
# writeFile4("D:/projects/finalProject/nifti/1/image3.vtk", "image")

In [ ]:
img.affine

array([[  -0.93300003,    0.        ,    0.        ,  252.96670532],
       [   0.        ,   -0.93300003,    0.        ,  238.96670532],
       [   0.        ,    0.        ,    5.        , -420.        ],
       [   0.        ,    0.        ,    0.        ,    1.        ]])

In [ ]:
n = 0
for n in range(85):
    data = [ ]
    for y in range(512):
        dy = [ ]
        for x in range(512):
            dy.append(d[x][y][n])
        data.append(dy)
    plt.imshow(data)
    plt.show()

# d[0][0][n]

In [ ]:
d2 = img2.get_data()
n = 0
for n in range(85):
    data = [ ]
    for y in range(512):
        dy = [ ]
        for x in range(512):
            dy.append(d2[x][y][n])
        data.append(dy)
    plt.imshow(data)
    plt.show()

In [ ]:
txMin = 511
txMax = 0
tyMin = 511
tyMax = 0
tzMin = 84
tzMax = 0
findFor = 3
for z in range(zn):
    for y in range(yn):
        for x in range(xn):
            if d2[x][y][z] == findFor and txMin > x:
                txMin = x
            if d2[x][y][z] == findFor and txMax < x:
                txMax = x
            if d2[x][y][z] == findFor and tyMin > y:
                tyMin = y
            if d2[x][y][z] == findFor and tyMax < y:
                tyMax = y
            if d2[x][y][z] == findFor and tzMin > z:
                tzMin = z
            if d2[x][y][z] == findFor and tzMax < z:
                tzMax = z
print(txMin, txMax)
print(tyMin, tyMax)
print(tzMin, tzMax)

218 267
265 315
54 58


In [ ]:
findFor = 3
tData = [ ]
for z in range(zn):
    for y in range(yn):
        for x in range(xn):
            if d2[x][y][z] == findFor:
                tData.append(d[x][y][z])
print(tData)

[9.0, -8.0, -11.0, -13.0, -16.0, -13.0, -7.0, 5.0, 0.0, -13.0, -29.0, -35.0, -43.0, -59.0, -75.0, -85.0, -78.0, -69.0, -56.0, -39.0, -25.0, -15.0, -11.0, -5.0, -8.0, -38.0, -68.0, -99.0, -126.0, -149.0, -174.0, -215.0, -252.0, -288.0, -305.0, -291.0, -229.0, -139.0, -69.0, -26.0, -9.0, -2.0, 1.0, 5.0, 9.0, 7.0, -16.0, -57.0, -118.0, -183.0, -258.0, -325.0, -370.0, -421.0, -481.0, -539.0, -591.0, -632.0, -624.0, -527.0, -359.0, -215.0, -117.0, -48.0, -10.0, -1.0, -2.0, 0.0, 4.0, 12.0, 11.0, -13.0, -60.0, -126.0, -215.0, -317.0, -423.0, -517.0, -596.0, -664.0, -736.0, -799.0, -840.0, -855.0, -849.0, -802.0, -694.0, -543.0, -370.0, -215.0, -110.0, -51.0, -26.0, -17.0, -5.0, 7.0, 13.0, 17.0, 19.0, -28.0, -105.0, -214.0, -336.0, -458.0, -575.0, -674.0, -763.0, -826.0, -874.0, -904.0, -927.0, -936.0, -934.0, -915.0, -875.0, -809.0, -688.0, -529.0, -362.0, -230.0, -151.0, -97.0, -54.0, -27.0, -5.0, 18.0, 26.0, 49.0, 34.0, -32.0, -142.0, -282.0, -422.0, -545.0, -662.0, -770.0, -854.0, -910.0, 

In [ ]:
def unique(list1):
    uniqueList = [ ]
    tMin = float("inf")
    tMax = float("-inf")
    seqMin = list1[0]
    seqMax = list1[0]
    seqList = [ ]
    for x in list1:
        if x not in uniqueList:
            uniqueList.append(x)
        if x < tMin:
            tMin = x
        if x > tMax:
            tMax = x
        if (seqMin - 1) <= x and (seqMax + 1) >= x and seqMin > x:
            seqMin = x
        elif (seqMin - 1) <= x and (seqMax + 1) >= x and seqMax < x:
            seqMax = x
        elif (seqMin - 1) <= x and (seqMax + 1) >= x:
            continue
        else:
            seqList.append((seqMin, seqMax, seqMax - seqMin))
            seqMin = x
            seqMax = x
    return [uniqueList, tMin, tMax, seqList]
tData2 = unique(tData)

In [ ]:
tData2[3]

In [ ]:
for i in tData2[3]:
    if i[2] > 0:
        print(i)